tt_metal/llrt/tt_cluster.hpp

```diff
@@ -264,28 +265,28 @@ class Cluster {
 
     // There is one device driver per PCIe card. This map points id of the MMIO device points to the associated device
     // driver
-    std::unordered_map<chip_id_t, std::unique_ptr<tt_device>> mmio_device_id_to_driver_;
+    tt::IndexMap<std::unique_ptr<tt_device>> mmio_device_id_to_driver_;
 
     // Need to hold reference to cluster descriptor to detect total number of devices available in cluster
     // UMD static APIs `detect_available_device_ids` and `detect_number_of_chips` only returns number of MMIO mapped
     // devices
     std::string cluster_desc_path_;
     std::unique_ptr<tt_ClusterDescriptor> cluster_desc_;
     // There is an entry for every device that can be targeted (MMIO and remote)
-    std::unordered_map<chip_id_t, metal_SocDescriptor> sdesc_per_chip_;
+    tt::IndexMap<metal_SocDescriptor> sdesc_per_chip_;
```

Comment:
there will always be one soc desc per chip so this can just be a vector sized to total number of chips in a cluster